plugins/@grouparoo/pipedrive/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,139 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { PipedriveClient } from "../client";
+import { connect } from "../connect";
+import {
+  PipedriveCacheData,
+  getKnownPersonFieldMap,
+} from "./destinationMappingOptions";
+import { getGroupFieldKey } from "./listMethods";
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appId,
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+  const cacheData: PipedriveCacheData = { appId, appOptions };
+
+  const newEmail = newProfileProperties["Email"];
+  const oldEmail = oldProfileProperties["Email"];
+
+  const newFoundId = await client.findPersonIdByEmail(newEmail);
+
+  let oldFoundId = null;
+  if (oldEmail && oldEmail !== newEmail) {
+    oldFoundId = await client.findPersonIdByEmail(oldEmail);
+  }
+
+  const foundId = oldFoundId || newFoundId;
```

Comment:
The nuance in the other ones tends to be this:
* for deleting it's `oldFoundId || newFoundId` but for updating it's `newFoundId || oldFoundId`

Reply:
Hmm, I still think it should be `oldFoundId || newFoundId`. If you're updating the email/foreignkey, I believe you would want to search for the old email to be able to update it to a new one.

Reply:
To document this here a bit, Brian is right in that there _is_ a difference between deleting and updating. The special case I wasn't seeing comes to light when dealing with profiles that may have been created from other sources (not grouparoo). 

Essentially, if you're **deleting** someone and changing their email at the same time _and_ a user with that new email already exists in the destination, we want to delete the old user and leave the one with the new email alone.

However, if you're **updating** someone and changing their email at the same time _and_ a user with that new email already exists in the destination, we would want to update the new one instead of the old one. This is mainly because if we were to update the old one at this point, we could create a unique key conflict by now having two people with the same (new) email. Note that in this case, the profile with the old email would remain untouched.

I'll write more about this and create a document to standardize the way Grouparoo handles primary id changes, but wanted to leave a note here since this is where it came up.